test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/NetCorePackageReferenceProjectTests.cs

```diff
@@ -2316,6 +2316,694 @@ public async Task GetInstalledPackages_WithNominationUpdate_ReloadsCache()
+                    {
+                        packageSpec = packageSpec.WithTestProjectReference(prevPackageSpec);
+                    }
+
+                    // Restore info
+                    var projectRestoreInfo = ProjectJsonTestHelpers.GetDGSpecFromPackageSpecs(packageSpec);
+                    projectCache.AddProjectRestoreInfo(projectNames, projectRestoreInfo, new List<IAssetsLogMessage>());
+                    projectCache.AddProject(projectNames, projectAdapter, project).Should().BeTrue();
+                    prevProj = projectFullPath;
+                    packageSpecs[i] = packageSpec;
+                    prevPackageSpec = packageSpec;
+                    projectFullPaths[i] = projectFullPath;
+                }
+
+                for (int i = 0; i < numberOfProjects; i++)
+                {
+                    // Install packageB since packageA is already there.
+                    await nuGetPackageManager.InstallPackageAsync(netCorePackageReferenceProjects[i], packageB, new ResolutionContext(), new TestNuGetProjectContext(),
+                            sourceRepositoryProvider.GetRepositories(), sourceRepositoryProvider.GetRepositories(), CancellationToken.None);
+
+                    // This code added because nuGetPackageManager.InstallPackageAsync doesn't do updating ProjectSystemCache
+                    var installed = new LibraryDependency
+                    {
+                        LibraryRange = new LibraryRange(
+                            name: packageB.Id,
+                            versionRange: new VersionRange(packageB.Version),
+                            typeConstraint: LibraryDependencyTarget.Package),
+                    };
+
+                    var packageSpec = packageSpecs[i];
+                    packageSpec.TargetFrameworks.First().Dependencies.Add(installed);
+                    var projectRestoreInfo = ProjectJsonTestHelpers.GetDGSpecFromPackageSpecs(packageSpec);
+                    var projectNames = GetTestProjectNames(projectFullPaths[i], $"project{i}");
+                    projectCache.AddProjectRestoreInfo(projectNames, projectRestoreInfo, new List<IAssetsLogMessage>());
+                }
+
+                var initialInstalledPackages = (await netCorePackageReferenceProjects[numberOfProjects - 1].GetInstalledPackagesAsync(CancellationToken.None)).ToList();
+
+                // Act
+                var actions = await nuGetPackageManager.PreviewProjectsUninstallPackageAsync(
+                    netCorePackageReferenceProjects, // All projects
+                    packageB.Id,
+                    uninstallationContext,
+                    testNuGetProjectContext,
+                    CancellationToken.None);
+                await nuGetPackageManager.ExecuteNuGetProjectActionsAsync(
+                    netCorePackageReferenceProjects,
+                    actions,
+                    testNuGetProjectContext,
+                    new SourceCacheContext(),
```

Comment:
`SourceCacheContext` is disposable.  Wrap usage with a `using` statement.

This feedback applies globally (e.g.:  other tests).

Reply:
Addressed.